src/main/java/io/zipcoder/Classroom.java

```diff
@@ -44,69 +71,95 @@ public boolean addStudent(Student student) {
+        Arrays.sort(this.students, averageScoresThenLexigraphically);
+        return students;
     }
 
-    /**
-     * Return a string of student names and their averages, like.
-     *
-     * Students:
-     * Bart Simpson -> 72.4
-     * Homer Simpson -> 0.1
-     * Lisa Simpson -> 100.0
-     * Milhouse Van Houten -> 87.6
-     *
-     * If there are no students in the array, return No Students.
-     *
-     * @return
-     */
-    public String getClassScores(){
-        return null;
+    public TreeMap<String, ArrayList<String>> getGradeBook() {
+        Student[] listOfStudentsToBePlaceed = getStudentsByScore();
+        int aGrade = (int) Math.ceil(((double) listOfStudentsToBePlaceed.length) * 0.10);
+        int bGrade = (int) Math.ceil(((double) listOfStudentsToBePlaceed.length) * 0.29);
+        int cGrade = (int) Math.ceil(((double) listOfStudentsToBePlaceed.length) * 0.50);
+        int dGrade = (int) Math.ceil(((double) listOfStudentsToBePlaceed.length) * 0.89);
+        int fGrade = (int) Math.ceil(((double) listOfStudentsToBePlaceed.length) * 1);
+        TreeMap<String, ArrayList<String>> gradeBook = new TreeMap<String, ArrayList<String>>();
+        ArrayList<String> aStudents = new ArrayList<String>();
+        ArrayList<String> bStudents = new ArrayList<String>();
+        ArrayList<String> cStudents = new ArrayList<String>();
+        ArrayList<String> dStudents = new ArrayList<String>();
+        ArrayList<String> fStudents = new ArrayList<String>();
+        for(int i = 0; i < listOfStudentsToBePlaceed.length; i++){
+            if(i <= aGrade){
+                aStudents.add(listOfStudentsToBePlaceed[i].getFirstName() + " " + listOfStudentsToBePlaceed[i].getLastName());
+            } else if(i > aGrade && i <= bGrade){
+                bStudents.add(listOfStudentsToBePlaceed[i].getFirstName() + " " + listOfStudentsToBePlaceed[i].getLastName());
+            } else if(i > bGrade && i <= cGrade){
+                cStudents.add(listOfStudentsToBePlaceed[i].getFirstName() + " " + listOfStudentsToBePlaceed[i].getLastName());
+            } else if(i > cGrade && i <= dGrade){
+                dStudents.add(listOfStudentsToBePlaceed[i].getFirstName() + " " + listOfStudentsToBePlaceed[i].getLastName());
+            } else if (i <= fGrade && i > dGrade){
+                fStudents.add(listOfStudentsToBePlaceed[i].getFirstName() + " " + listOfStudentsToBePlaceed[i].getLastName());
+            }
+        }
+        gradeBook.put("A", aStudents);
+        gradeBook.put("B", bStudents);
+        gradeBook.put("C", cStudents);
+        gradeBook.put("D", dStudents);
+        gradeBook.put("F", fStudents);
+
+        return gradeBook;
     }
 
-    /**
-     * Sorts the Students array from highest average to lowest, and ties are broken alphabetically.
-     */
-    public void sortStudentsByScore() {
 
-    }
+    //comparator
+    public static Comparator<Student> averageScoresThenLexigraphically = new Comparator<Student>() {
```

Comment:
you should add "implements" Comparable to the class, the compiler can help check your work.